tests/components/intellifire/test_config_flow.py

```diff
@@ -1,55 +1,205 @@
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
     assert result["type"] == RESULT_TYPE_FORM
-    assert result["errors"] is None
+    assert result["errors"] == {}
+    assert result["step_id"] == "manual_device_entry"
 
     result2 = await hass.config_entries.flow.async_configure(
         result["flow_id"],
         {
-            "host": "1.1.1.1",
+            CONF_HOST: "1.1.1.1",
         },
     )
     await hass.async_block_till_done()
-
     assert result2["type"] == RESULT_TYPE_CREATE_ENTRY
-    assert result2["title"] == "Fireplace"
-    assert result2["data"] == {"host": "1.1.1.1"}
+    assert result2["title"] == "Fireplace 12345"
+    assert result2["data"] == {CONF_HOST: "1.1.1.1"}
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_single_discovery(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    mock_intellifire_config_flow: MagicMock,
+) -> None:
+    """Test single firedplace UDP discovery."""
+    with patch(
+        "intellifire4py.udp.AsyncUDPFireplaceFinder.search_fireplace",
+        return_value=["192.168.1.69"],
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
 
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"], {CONF_HOST: "192.168.1.69"}
+    )
+    await hass.async_block_till_done()
+    print("Result:", result)
+
+    assert result2["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result2["title"] == "Fireplace 12345"
+    assert result2["data"] == {CONF_HOST: "192.168.1.69"}
     assert len(mock_setup_entry.mock_calls) == 1
 
 
-async def test_form_cannot_connect(
-    hass: HomeAssistant, mock_intellifire_config_flow: MagicMock
+async def test_manual_entry(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    mock_intellifire_config_flow: MagicMock,
+) -> None:
+    """Test for multiple firepalce discovery - involing a pick_device step."""
+    with patch(
+        "intellifire4py.udp.AsyncUDPFireplaceFinder.search_fireplace",
+        return_value=["192.168.1.69", "192.168.1.33", "192.168.169"],
+    ):
+
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_USER}
+        )
+
+    assert result["step_id"] == "pick_device"
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={CONF_HOST: MANUAL_ENTRY_STRING}
+    )
+
+    await hass.async_block_till_done()
+    assert result2["step_id"] == "manual_device_entry"
+
+
+async def test_mutli_discovery(
+    hass: HomeAssistant,
+    mock_setup_entry: AsyncMock,
+    mock_intellifire_config_flow: MagicMock,
+) -> None:
+    """Test for multiple firepalce discovery - involing a pick_device step."""
+    with patch(
+        "intellifire4py.udp.AsyncUDPFireplaceFinder.search_fireplace",
```

Comment:
There are few more patch targets that could be cleaned up to only patch where it is needed.